internal/snmp/config.go

```diff
@@ -0,0 +1,34 @@
+package snmp
+
+import (
+	"github.com/influxdata/telegraf/internal"
+)
+
+type ClientConfig struct {
+	// Timeout to wait for a response.
+	Timeout internal.Duration `toml:"timeout"`
+	Retries int               `toml:"retries"`
+	// Values: 1, 2, 3
+	Version uint8 `toml:"version"`
+
+	// Parameters for Version 1 & 2
+	Community string `toml:"community"`
+
+	// Parameters for Version 2 & 3
+	MaxRepetitions uint8 `toml:"max_repetitions"`
+
+	// Parameters for Version 3
+	ContextName string `toml:"context_name"`
+	// Values: "noAuthNoPriv", "authNoPriv", "authPriv"
+	SecLevel string `toml:"sec_level"`
+	SecName  string `toml:"sec_name"`
+	// Values: "MD5", "SHA", "". Default: ""
+	AuthProtocol string `toml:"auth_protocol"`
+	AuthPassword string `toml:"auth_password"`
+	// Values: "DES", "AES", "". Default: ""
+	PrivProtocol string `toml:"priv_protocol"`
+	PrivPassword string `toml:"priv_password"`
+	EngineID     string `toml:"-"`
+	EngineBoots  uint32 `toml:"-"`
+	EngineTime   uint32 `toml:"-"`
```

Comment:
This and everything in internal/snmp came from the snmp input plugin.  When I moved it into internal I noticed it has ugly spots like here and in your other review comments.  Since the integration tests are lacking, there is real potential to break stuff without knowing.  (The snmp input unit tests mock all network traffic.  It also mocks running the snmptranslate cli tool.)  In this PR I avoided modifying gosnmpwrapper because of the risk.

Cleaning up the snmp input is a good idea but it's not something we need to do for the ifname processor.